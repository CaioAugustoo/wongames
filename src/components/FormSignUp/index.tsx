import Link from 'next/link'
import { AccountCircle, Email, Lock } from '@styled-icons/material-outlined'

import { FormWrapper, FormLink } from 'components/Form'
import Button from 'components/Button'
import TextField from 'components/TextField'

const FormSignUp = () => (
  <FormWrapper>
    <form>
      <TextField
        name="name"
        placeholder="Nome"
        type="name"
        icon={<AccountCircle />}
      />
      <TextField
        name="email"
        placeholder="Email"
        type="email"
        icon={<Email />}
      />
      <TextField
        name="password"
        placeholder="Senha"
        type="password"
        icon={<Lock />}
      />
      <TextField
        name="confirm-password"
        placeholder="Confirmar senha"
        type="password"
        icon={<Lock />}
      />
      <Button size="large" fullWidth>
        Cadastrar agora!
      </Button>

      <FormLink>
        JÃ¡ possui uma conta?{' '}
        <Link href="/sign-in">
          <a>Entre agora</a>
        </Link>
      </FormLink>
    </form>
  </FormWrapper>
)

export default FormSignUp
