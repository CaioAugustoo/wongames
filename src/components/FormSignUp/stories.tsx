import { Story, Meta } from '@storybook/react/types-6-0'
import FormSignUp from '.'

export default {
  title: 'Form/FormSignUp',
  component: FormSignUp
} as Meta

export const Default: Story = () => (
  <div style={{ width: 300, margin: 'auto' }}>
    <FormSignUp />
  </div>
)
