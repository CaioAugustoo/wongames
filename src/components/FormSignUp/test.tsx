import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import FormSignUp from '.'

describe('<FormSignUp />', () => {
  it('should render the form', () => {
    const { container } = renderWithTheme(<FormSignUp />)

    expect(screen.getByPlaceholderText(/name/i)).toBeInTheDocument()
    expect(screen.getByPlaceholderText(/email/i)).toBeInTheDocument()
    expect(screen.getByPlaceholderText('Password')).toBeInTheDocument()
    expect(screen.getByPlaceholderText('Confirm password')).toBeInTheDocument()
    expect(
      screen.getByRole('button', { name: /sign up now/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render text and link to sign in', () => {
    renderWithTheme(<FormSignUp />)

    expect(screen.getByRole('link', { name: /sign in/i })).toBeInTheDocument()
    expect(screen.getByText(/already have an account\?/i)).toBeInTheDocument()
  })
})
