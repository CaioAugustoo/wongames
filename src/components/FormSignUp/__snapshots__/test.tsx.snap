// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormSignUp /> should render the form 1`] = `
.c4 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background: #EAEAEA;
  border-radius: 0.5rem;
  padding: 0 1.6rem;
  border: 0.1rem solid;
  border-color: #EAEAEA;
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
}

.c2:focus-within {
  border: 0.1rem solid #47C8FF;
}

.c5 {
  color: #030517;
  font-family: Poppins,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  font-size: 1.6rem;
  padding: 0.8rem;
  padding-left: 0.8rem;
  background: transparent;
  border: 0;
  outline: none;
  width: 100%;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 2.2rem;
  color: #8F8F8F;
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
}

.c3 > svg {
  width: 100%;
}

.c7 {
  background: #47c8ff;
  color: #FAFAFA;
  border: none;
  border-radius: 0.4rem;
  padding: 0.8rem;
  cursor: pointer;
  outline: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
  height: 5rem;
  font-size: 1.6rem;
  padding: 0.8rem 4.8rem;
  width: 100%;
}

.c7:hover {
  background: #2499ae;
}

.c0 .c1 {
  margin: 0.8rem 0;
}

.c0 .c6 {
  margin: 3.2rem auto 1.6rem;
}

.c8 {
  font-size: 1.4rem;
  color: #030517;
  text-align: center;
}

.c8 a {
  color: #47C8FF;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-bottom: 0.1rem solid #47C8FF;
  -webkit-transition: color,border,0.1s ease-in-out;
  transition: color,border,0.1s ease-in-out;
}

.c8 a:hover {
  color: #14b9ff;
  border-bottom: 0.1rem solid #14b9ff;
}

<div
  class="c0"
>
  <form>
    <div
      class="c1 "
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <svg
            aria-hidden="true"
            class="c4"
            fill="currentColor"
            focusable="false"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0V0z"
              fill="none"
            />
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zM7.07 18.28c.43-.9 3.05-1.78 4.93-1.78s4.51.88 4.93 1.78C15.57 19.36 13.86 20 12 20s-3.57-.64-4.93-1.72zm11.29-1.45c-1.43-1.74-4.9-2.33-6.36-2.33s-4.93.59-6.36 2.33A7.95 7.95 0 014 12c0-4.41 3.59-8 8-8s8 3.59 8 8c0 1.82-.62 3.49-1.64 4.83zM12 6c-1.94 0-3.5 1.56-3.5 3.5S10.06 13 12 13s3.5-1.56 3.5-3.5S13.94 6 12 6zm0 5c-.83 0-1.5-.67-1.5-1.5S11.17 8 12 8s1.5.67 1.5 1.5S12.83 11 12 11z"
            />
          </svg>
        </div>
        <input
          class="c5"
          name="name"
          placeholder="Nome"
          type="name"
          value=""
        />
      </div>
    </div>
    <div
      class="c1 "
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <svg
            aria-hidden="true"
            class="c4"
            fill="currentColor"
            focusable="false"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0V0z"
              fill="none"
            />
            <path
              d="M22 6c0-1.1-.9-2-2-2H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6zm-2 0l-8 5-8-5h16zm0 12H4V8l8 5 8-5v10z"
            />
          </svg>
        </div>
        <input
          class="c5"
          name="email"
          placeholder="Email"
          type="email"
          value=""
        />
      </div>
    </div>
    <div
      class="c1 "
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <svg
            aria-hidden="true"
            class="c4"
            fill="currentColor"
            focusable="false"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              fill="none"
            >
              <path
                d="M0 0h24v24H0V0z"
              />
              <path
                d="M0 0h24v24H0V0z"
                opacity="0.87"
              />
            </g>
            <path
              d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
            />
          </svg>
        </div>
        <input
          class="c5"
          name="password"
          placeholder="Senha"
          type="password"
          value=""
        />
      </div>
    </div>
    <div
      class="c1 "
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <svg
            aria-hidden="true"
            class="c4"
            fill="currentColor"
            focusable="false"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              fill="none"
            >
              <path
                d="M0 0h24v24H0V0z"
              />
              <path
                d="M0 0h24v24H0V0z"
                opacity="0.87"
              />
            </g>
            <path
              d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
            />
          </svg>
        </div>
        <input
          class="c5"
          name="confirm-password"
          placeholder="Confirmar senha"
          type="password"
          value=""
        />
      </div>
    </div>
    <button
      class="c6 c7"
    >
      <span>
        Cadastrar agora!
      </span>
    </button>
    <div
      class="c8"
    >
      JÃ¡ possui uma conta?
       
      <a
        href="/sign-in"
      >
        Entre agora
      </a>
    </div>
  </form>
</div>
`;
